pgdrive/world/force_fps.py

```diff
@@ -20,20 +21,25 @@ def __init__(self, fps, start=False):
     def interval(self):
         return 1 / self.fps if self.fps is not None else None
 
-    def __enter__(self):
+    def tick(self):
         # print("Force fps, now: ", self.last)
-        now = time.time()
-        if self.interval and now - self.last < self.interval:
-            time.sleep(self.interval - (now - self.last))
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        self.last = time.time()
+        sim_interval = self.pg_world.taskMgr.globalClock.getDt()
+        if self.interval and sim_interval < self.interval:
+            time.sleep(self.interval - sim_interval)
 
     def toggle(self):
         if self.state == self.UNLIMITED:
+            self.pg_world.taskMgr.add(self.force_fps_task, "force_fps")
             self.state = self.FORCED
             self.fps = self.init_fps
-
         elif self.state == self.FORCED:
+            self.pg_world.taskMgr.remove("force_fps")
             self.state = self.UNLIMITED
             self.fps = None
+
+    def force_fps_task(self, task):
+        self.tick()
+        return task.cont
+
+    def __call__(self, other_condition):
+        return self.state == self.FORCED and self.pg_world.mode == RENDER_MODE_ONSCREEN and other_condition
```

Comment:
Stop using __call__ if you only mean for returning a boolean flag.

Create a more informative function name for this.

Reply:
I will resolve this